yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -215,8 +223,10 @@ private[spark] class Client(
     // and add them as local resources to the application master.
     val fs = FileSystem.get(hadoopConf)
     val dst = new Path(fs.getHomeDirectory(), appStagingDir)
-    val nns = getNameNodesToAccess(sparkConf) + dst
-    obtainTokensForNamenodes(nns, hadoopConf, credentials)
+    val nns =
+      SparkHadoopUtil.get.asInstanceOf[YarnSparkHadoopUtil].getNameNodesToAccess(sparkConf) + dst
```

Comment:
What about my suggestion of having a `YarnSparkHadoopUtil.get` instead of this cast everywhere?